expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -258,6 +262,25 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                         // treat as plain content
                         continue;
                     }
+                    if (flags.contains(Flag.STRICT_ESCAPES)) {
+                        if (!itr.hasNext()) {
```

Comment:
This should all be done within the next state. Just as we don't want to add another state flag, we also don't want to introduce backtracking for a similar reason: the state machine should only have one way to reason about it if possible.

This logic should all fit into the `case '$'` substate below.

Reply:
Yes, I wanted to minimize changes to the current code, but I'll do that.